### PR TITLE
#120 - Added IsHeader matcher

### LINKED ISSUE #120: Add header matcher
It is quite common it tests to check headers coming from slice. There is no convenient matcher for it in Hamcrest out of the box or in our library

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #120 
Added IsHeader matcher to match response headers along with tests

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#121](https://github.com/artipie/http/issues/121) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/13636202-b7af-4540-8d03-f88b701c9cac); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/http/hm/IsHeaderTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        final String value = "100";
+        final IsHeader matcher = new IsHeader(name, value);
+        MatcherAssert.assertThat(
+            matcher.matches(new MapEntry<>(name, value)),
+            new IsEqual<>(true)
+        );
+    }
+
+    @Test
+    void shouldMatchNameMatching() {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/http/hm/IsHeaderTest.java:51]: @olegmoz we are asserting using matcher for a value not for name, so I would rename the test to `shouldMatchUsingValueMatcher`

Comment (olegmoz) [reply src/test/java/com/artipie/http/hm/IsHeaderTest.java]: @Vatavuk thank you for noticing, that is weird name indeed! Fixed

**Comment by 0crat** [comment]
Job was finished in 19 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**